cmd/nodecmd/create.go

```diff
@@ -151,284 +140,126 @@ func printNoCredentialsOutput() {
+				publicIPMap[node] = cloudConfig.PublicIPs[i]
 			}
-			ux.Logger.PrintToUser(fmt.Sprintf("AWS cloud server instance %s stopped", instanceID))
 		}
-		if len(failedNodes) > 0 {
-			ux.Logger.PrintToUser("Failed nodes: ")
-			for i, node := range failedNodes {
-				ux.Logger.PrintToUser(fmt.Sprintf("Failed to stop node %s due to %s", node, nodeErrors[i]))
+	} else {
+		// Get GCP Credential, zone, Image ID, service account key file path, and GCP project name
+		gcpClient, zone, imageID, credentialFilepath, projectName, err := getGCPConfig()
+		if err != nil {
+			return err
+		}
+		cloudConfig, err = createGCPInstance(usr, gcpClient, zone, imageID, credentialFilepath, projectName)
+		if err != nil {
+			return err
+		}
+		if !useStaticIP {
+			publicIPMap, err = gcpAPI.GetInstancePublicIPs(gcpClient, projectName, zone, cloudConfig.InstanceIDs)
+			if err != nil {
+				return err
+			}
+		} else {
+			for i, node := range cloudConfig.InstanceIDs {
+				publicIPMap[node] = cloudConfig.PublicIPs[i]
 			}
-			ux.Logger.PrintToUser("Stop the above instance(s) on AWS console to prevent charges")
-			return fmt.Errorf("failed to stop node(s) %s", failedNodes)
 		}
-		return err
+		gcpProjectName = projectName
+		gcpCredentialFilepath = credentialFilepath
 	}
-	if err := createClusterNodeConfig(instanceIDs, elasticIPs, region, ami, keyPairName, certFilePath, securityGroupName, clusterName); err != nil {
+	if err = createClusterNodeConfig(cloudConfig, clusterName, cloudService); err != nil {
 		return err
 	}
+	if cloudService == constants.GCPCloudService {
+		if err = updateClusterConfigGCPKeyFilepath(gcpProjectName, gcpCredentialFilepath); err != nil {
+			return err
+		}
+	}
 	err = terraform.RemoveDirectory(app.GetTerraformDir())
 	if err != nil {
 		return err
 	}
-	publicIPMap := map[string]string{}
-	if !useEIP {
-		publicIPMap, err = awsAPI.GetInstancePublicIPs(ec2Svc, instanceIDs)
-		if err != nil {
-			return err
-		}
-	} else {
-		for i, node := range instanceIDs {
-			publicIPMap[node] = elasticIPs[i]
-		}
-	}
 	inventoryPath := app.GetAnsibleInventoryDirPath(clusterName)
-	if err := ansible.CreateAnsibleHostInventory(inventoryPath, certFilePath, publicIPMap); err != nil {
+	if err = ansible.CreateAnsibleHostInventory(inventoryPath, cloudConfig.CertFilePath, cloudService, publicIPMap); err != nil {
 		return err
 	}
-	time.Sleep(15 * time.Second)
+	time.Sleep(30 * time.Second)
 
 	avalancheGoVersion, err := getAvalancheGoVersion()
 	if err != nil {
 		return err
 	}
-	ux.Logger.PrintToUser("Installing AvalancheGo and Avalanche-CLI and starting bootstrap process on the newly created EC2 instance(s)...")
-	if err := runAnsible(inventoryPath, avalancheGoVersion, clusterName); err != nil {
+	ux.Logger.PrintToUser("Installing AvalancheGo and Avalanche-CLI and starting bootstrap process on the newly created Avalanche node(s) ...")
+	ansibleHostIDs := []string{}
+	for _, instanceID := range cloudConfig.InstanceIDs {
+		var ansibleHostName string
```

Comment:
we probably need a function to encapsulate this: instanceid + cloud service -> ansible host name, ansible host name -> instanceid + cloud service